ecr/base.go

```diff
@@ -50,36 +51,64 @@ type ecrAPI interface {
 	PutImageWithContext(aws.Context, *ecr.PutImageInput, ...request.Option) (*ecr.PutImageOutput, error)
 }
 
-func (b *ecrBase) getManifest(ctx context.Context) (*ecr.Image, error) {
-	imageIdentifier := b.ecrSpec.ImageID()
-	log.G(ctx).WithField("imageIdentifier", imageIdentifier).Debug("ecr.base.manifest: getting image")
+// getImageByDescriptor retrieves an image from ECR for a given OCI descriptor.
+func (b *ecrBase) getImageByDescriptor(ctx context.Context, desc ocispec.Descriptor) (*ecr.Image, error) {
+	input := ecr.BatchGetImageInput{
+		ImageIds: []*ecr.ImageIdentifier{
+			&ecr.ImageIdentifier{ImageDigest: aws.String(desc.Digest.String())},
+		},
+	}
+	if desc.MediaType != "" {
+		input.AcceptedMediaTypes = []*string{aws.String(desc.MediaType)}
+
+	}
 
-	batchGetImageInput := &ecr.BatchGetImageInput{
-		RegistryId:     aws.String(b.ecrSpec.Registry()),
-		RepositoryName: aws.String(b.ecrSpec.Repository),
-		ImageIds:       []*ecr.ImageIdentifier{imageIdentifier},
+	imgs, err := b.runGetImage(ctx, input)
+	if err != nil {
+		return nil, err
 	}
-	log.G(ctx).WithField("batchGetImageInput", batchGetImageInput).Trace("ecr.base.manifest")
+	return imgs[0], nil
+}
 
-	batchGetImageOutput, err := b.client.BatchGetImageWithContext(ctx, batchGetImageInput)
+func (b *ecrBase) getImage(ctx context.Context) (*ecr.Image, error) {
```

Comment:
nit: can you move this above `getImageByDescriptor` as it'll be called first in most workflows?

Reply:
Wait a sec.. wasn't it caller above callee?

Reply:
Caller above callee, but `getImage` and `getImageByDescriptor` don't have a caller-callee relationship.  Instead, they both call `runGetImage` which is below both.

Reply:
Ah, yep, my brain doesn't run git well.

Reply:
Should be updated and reflected in the latest push.